### ISSUE TITLE
Does CNTK support validation-based early stopping?

### ISSUE THREAD

**Issue by MaggieWang1011** [Microsoft/CNTK#633]
I want to stop training once the least mean square error is minimized on the validation database, which consists of different data from the training database. Does CNTK support this? I read the tutorial carefully, but I could find something reaching this. Is there a tutorial about this?
Thank you very much.

**Comment by dongyu888** [comment]
Yes, if you specify the dev set in the config and use auto learning rate adjustment. Check the SGD section of the wiki.

**Comment by CPostelnicu** [comment]
Hi, 
I don't understand how the validation set can be specified within the train block, together with the details of SGD. I have checked the [documentation](https://github.com/Microsoft/CNTK/wiki/SGD-block) and, in my understanding:

- within the SGD block we can specify:  _modelPath: the full path to save the final model. Must be provided and points to a valid file name._ 
- within the train block, but outside the SGD block, a reader must be specified that should be pointing to the train set. 

How / Where could one specify that the early stopping criterion should be computed on the validation dataset (i.e. a different dataset from the one the training uses)? 

Thank you!

**Comment by dongyu888** [comment]
You already have a reader section in your config, right? You just need to create a cvreader section in parallel to the reader section to provide information on the validation set.

**Comment by CPostelnicu** [comment]
Thank you very much!

**Comment by oxrider** [comment]
@dongyu888 can you provide a short example for this? I get the idea but don't know how to start.

**Comment by cha-zhang** [comment]
https://github.com/Microsoft/CNTK/blob/master/Examples/Image/Classification/GoogLeNet/InceptionV3/BrainScript/InceptionV3.cntk

**Comment by MiloLT** [comment]
@dongyu888 i was wondering if there is any C# snippets for CrossValidation? what i like to do is to evaluate the [Validation] error for each iteration of the trainer.
